addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureFileInclusionScanRule.java

```diff
@@ -547,4 +516,39 @@ private int calcLengthMatchPercentage(int a, int b) {
 
         return (int) ((((double) Math.min(a, b)) / Math.max(a, b)) * 100);
     }
+
+    private AlertBuilder createAlert(
+            String paramname,
+            String prefixedUrlfilename,
+            Integer randomversussourcefilenamematchpercentage,
+            HttpMessage sourceattackmsg,
+            String uri) {
+        return newAlert()
+                .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                .setDescription(Constant.messages.getString("ascanbeta.sourcecodedisclosure.desc"))
+                .setUri(uri)
+                .setParam(paramname)
+                .setAttack(prefixedUrlfilename)
+                .setOtherInfo(
+                        Constant.messages.getString(
+                                "ascanbeta.sourcecodedisclosure.lfibased.extrainfo",
+                                prefixedUrlfilename,
+                                NON_EXISTANT_FILENAME,
+                                randomversussourcefilenamematchpercentage,
+                                this.thresholdPercentage))
+                .setSolution(
+                        Constant.messages.getString("ascanbeta.sourcecodedisclosure.lfibased.soln"))
+                .setEvidence(
+                        Constant.messages.getString(
+                                "ascanbeta.sourcecodedisclosure.lfibased.evidence"))
+                .setMessage(sourceattackmsg);
+    }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        String exampleUri = "https://example.com";
+        return List.of(
+                createAlert("name", "../config/database.php", 48, getBaseMsg(), exampleUri).build(),
+                createAlert("name", "../secrets/passwords", 31, getBaseMsg(), exampleUri).build());
```

Comment:
- Return a single alert, they are the same.
 - Pass null instead of calling `getBaseMsg()`, that's not initialised when getting the example alerts.